shared/ui/src/components/ElementsBox/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { CSSProperties } from '@emotion/serialize';
+import { FlexBox, Padding, PaddingSize } from '../../layout';
+import { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+
+/**
+ * @paddingSize 버튼 셋의 패딩
+ * number : 상하좌우 패딩
+ * [number,number] : 상하, 좌우
+ * [number,number,number,number] : 상, 우, 하, 좌
+ * @param jusitfy : justify-content 속성
+ * @param leftElement : 좌측에 위치시킬 element
+ * @param rightElement : 우측에 위치시킬 element (기본값 : () => <></>)
+ * @param midElement : 중간에 위치시킬 element (기본값 : () => <></>)
+ */
+export interface ElementsBoxProps {
+  paddingSize: PaddingSize;
+  justify: CSSProperties['justifyContent'];
+  leftElement: () => EmotionJSX.Element;
+  rightElement?: () => EmotionJSX.Element;
+  midElement?: () => EmotionJSX.Element;
```

Comment:
```suggestion
  leftElement: ReactNode;
  rightElement?: ReactNode;
  midElement?: ReactNode;
```

함수가 아니라 컴포넌트 자체를 넘겨주니까 이제 `JSX.Element` 또는 `ReactNode`의 타입으로 넘길수있음..
근데 둘 중에 어떤게 더 적절한지 아직 잘 모르겠어서 찾아봤는데, ReactNode가 null이나 undefined까지도 포함하고 있다고해서 보통은 이걸 많이 쓰는것 같음 [참고](https://velog.io/@hanei100/ReactElement-vs-ReactNode-vs-JSX.Element)

```
export interface FlexBoxProps {
  align: CSSProperties['alignItems'];
  justify?: CSSProperties['justifyContent'];
  direction?: CSSProperties['flexDirection'];
  gap?: CSSProperties['gap'];
  children: ReactNode;
}
```
그럼 FlexBox의 children 타입도 JSX.Element가 아니라 ReactNode로 수정해야겠다!!